zerver/views/message_fetch.py

```diff
@@ -106,19 +110,40 @@ def get_messages_backend(
     *,
     anchor_val: Annotated[str | None, ApiParamConfig("anchor")] = None,
     include_anchor: Json[bool] = True,
-    num_before: Json[NonNegativeInt],
-    num_after: Json[NonNegativeInt],
+    num_before: Json[NonNegativeInt] = 0,
+    num_after: Json[NonNegativeInt] = 0,
     narrow: Json[list[NarrowParameter] | None] = None,
     use_first_unread_anchor_val: Annotated[
         Json[bool], ApiParamConfig("use_first_unread_anchor")
     ] = False,
     client_gravatar: Json[bool] = True,
     apply_markdown: Json[bool] = True,
+    client_requested_message_ids: Annotated[
+        Json[list[NonNegativeInt] | None], ApiParamConfig("message_ids")
+    ] = None,
 ) -> HttpResponse:
+    # User has to either provide message_ids or both num_before and num_after.
+    if (
+        num_before or num_after or anchor_val is not None or use_first_unread_anchor_val
+    ) and client_requested_message_ids is not None:
+        raise IncompatibleParametersError(
+            ["num_before", "num_after", "anchor", "message_ids", "use_first_unread_anchor_val"]
```

Comment:
I think this should be `use_first_unread_anchor`; the `_val` is a local variable detail, not part of the API.

Reply:
Oops! Fixed.